envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/config/LocalReplyConfigFactoryTest.kt

```diff
@@ -0,0 +1,524 @@
+                )
+            }
+    }
+
+    @Test
+    fun `should throw exception when more then one matcher is defined for HeaderMatcher`() {
+        // given
+        val properties = LocalReplyMapperProperties().apply {
+            enabled = true
+            matchers = listOf(
+                MatcherAndMapper().apply {
+                    headerMatcher = HeaderMatcher().apply {
+                        name = ":path"
+                        regexMatch = "regex_*"
+                        exactMatch = "exact"
+                    }
+                }
+            )
+        }
+
+        // expects
+        assertThatExceptionOfType(IllegalArgumentException::class.java)
+            .isThrownBy { LocalReplyConfigFactory(properties) }
+            .satisfies {
+                assertThat(it.message).isEqualTo(
+                    "Only one of: exactMatch, regexMatch can be defined."
+                )
+            }
+    }
+
+    @Test
+    fun `should throw exception when more then one property is defined in response format`() {
+        // given
+        val properties = LocalReplyMapperProperties().apply {
+            enabled = true
+            responseFormat = ResponseFormat().apply {
+                jsonFormat = mapOf(
+                    "body" to "custom response body"
+                )
+                textFormat = "custom response body"
+            }
+        }
+
+        // expects
+        assertThatExceptionOfType(IllegalArgumentException::class.java)
+            .isThrownBy { LocalReplyConfigFactory(properties) }
+            .satisfies {
+                assertThat(it.message).isEqualTo(
+                    "Only one of: jsonFormat, textFormat can be defined."
+                )
+            }
+    }
+
+    @Test
+    fun `should throw exception when more then one property is defined in specific matcher response format`() {
+        // given
+        val properties = LocalReplyMapperProperties().apply {
+            enabled = true
+            matchers = listOf(
+                MatcherAndMapper().apply {
+                    headerMatcher = HeaderMatcher().apply {
+                        name = ":path"
+                        regexMatch = "regex_*"
+                    }
+                    responseFormat = ResponseFormat().apply {
+                        textFormat = "custom response body"
+                        jsonFormat = mapOf(
+                            "body" to "custom response body"
+                        )
+                    }
+                }
+            )
+        }
+
+        // expects
+        assertThatExceptionOfType(IllegalArgumentException::class.java)
+            .isThrownBy { LocalReplyConfigFactory(properties) }
+            .satisfies {
+                assertThat(it.message).isEqualTo(
+                    "Only one of: jsonFormat, textFormat can be defined."
+                )
+            }
+    }
+
+    private val expectedConfigForResponseFlagsMatcher = """mappers {
```

Comment:
I visually prefer "mappers {" to be in the next line but if you like it this way then this can stay like this.